### ISSUE TITLE
v 2.5.0 does not recognize graphql resources 

### ISSUE THREAD

**Issue by BernardA** [api-platform/api-platform#1328]
**API Platform version(s) affected**: 2.5.0

**Description**  
This is linked to [this issue.](https://github.com/api-platform/api-platform/issues/1327)

Had the issue above which was solved by updating `api-platform/core` to version 2.5.0.

Now some of my Entity simply disappeared as resources or do not appear as Collection. 

A query like below, that worked fine before this change now throws error as below:

     query{
            categories {
                edges {
                    node {
                        id
                        title
                        parent {
                            id
                            title
                        }
                        root {
                            id
                            title
                        }
                    }
                }
                totalCount
            }
      }

Error throw: "Cannot query field \"edges\" on type \"Category\".",

For the entity Ad, the query below works fine:

               ads{
                   edges {
                       node {
                           id
                           _id
                           description
                           user {
                               id
                               username
                               image{
                                   filename
                               }
                               address{
                                   city
                               }
                           }
                           category {
                               id
                               title
                               parent{
                                   title
                               }
                           }
                           budget
                       }
                   }
                   totalCount
               }

The User entity with annotations as below simply no longer shows on `graphiql` neither for individual user nor for a Collection.

     * @ApiResource(
     *      itemOperations={
     *          "get"={
     *              "access_control"="is_granted('IS_AUTHENTICATED_FULLY')",
     *              "normalization_context"={
     *                  "groups"={"get"}
     *              }
     *          },
     *          "put"={
     *              "access_control"="is_granted('IS_AUTHENTICATED_FULLY') and object == user",
     *              "denormalization_context"={
     *                  "groups"={"put"}
     *              },
     *              "normalization_context"={
     *                  "groups"={"get"}
     *              }
     *          },
     *          "put-password-reset"={
     *              "access_control"="is_granted('IS_AUTHENTICATED_FULLY') and object == user",
     *              "method"="PUT",
     *              "path"="/users/{id}/password-reset",
     *              "controller"=PasswordResetAction::class,
     *              "denormalization_context"={
     *                  "groups"={"put-password-reset"}
     *              },
     *              "validation_groups"={"put-password-reset"}
     *          },
     *          "put-profile-change"={
     *              "access_control"="is_granted('IS_AUTHENTICATED_FULLY') and object == user",
     *              "method"="PUT",
     *              "path"="/users/{id}/profile-change",
     *              "controller"=ProfileChangeAction::class,
     *              "denormalization_context"={
     *                  "groups"={"put-profile-change"}
    *              },
    *              "validation_groups"={"put-profile-change"}
     *          },
     *          "put-password-recovery"={
    *              "method"="PUT",
     *              "path"="/users/{id}/password-recovery-reset",
    *              "controller"=PasswordRecoveryAction::class,
     *              "denormalization_context"={
     *                  "groups"={"put-password-recovery"}
    *              },
     *              "validation_groups"={"put-password-recovery"}
     *          },
     *          "put-register-social"={
     *              "method"="PUT",
     *              "path"="/users/{id}/register-social",
     *              "controller"=RegisterSocialAction::class,
     *              "denormalization_context"={
     *                  "groups"={"put-register-social"}
     *              },
     *              "validation_groups"={"put-register-social"}
     *          }
     *      },
     *      collectionOperations={
     *          "post"={
     *              "denormalization_context"={
     *                  "groups"={"post", "post-social-register-google"}
     *              },
     *              "normalization_context"={
     *                  "groups"={"get"}
     *              },
     *              "validation_groups"={"post", "post-social-register-google"}
     *          }
     *      },
     *      graphql={
     *          "create"={
     *              "normalization_context"={"groups"={"get"}},
     *              "denormalization_context"={"groups"={"post"}}
     *          },
     *          "query",
     *          "update",
     *          "delete",
     *      }
     * )

**How to reproduce**  
Full code at [this repository](https://github.com/BernardA/quiamo-api). 
**Possible Solution**  

**Additional Context**

**Comment by BernardA** [comment]
Is there anybody out there?
Apparently not. 
In any case I will give an update on this issue.

On the User entity, I removed the specific `graphql` annotation below and it is back as available as a Collection.
      
     *      graphql={
     *          "create"={
     *              "normalization_context"={"groups"={"get"}},
     *              "denormalization_context"={"groups"={"post"}}
     *          },
     *          "query",
     *          "update",
     *          "delete",
     *      }

It seems that I did not lose any other functionality with that change. Still to confirm.

Concerning the Categories entity, I am able to get a simple list with the query below. No `edges` or `totalCount` are available.

      query{
		categories{
                   title
                   root{
                     title
                  }
                  parent{
                    title
                  }
           }
     }

Let me remind you that both the User and Category entities were working fine with v `2.4.6`.

**Comment by dunglas** [comment]
Hi, and thanks for reporting? Are you able to find the specific commits that break things? Or could you provide a reproducer (a failing test or a minimal project reproducing the problem would be great). Thanks.

**Comment by mahmoodbazdar** [comment]
@BernardA It's because you are using `query` in graphql. In `2.5.0` its seperated to `item_query` and `collection_query`. 
you should change it to:
```
 *      graphql={
 *          "create"={
 *              "normalization_context"={"groups"={"get"}},
 *              "denormalization_context"={"groups"={"post"}}
 *          },
 *          "item_query",
 *          "collection_query",
 *          "update",
 *          "delete",
 *      }
```

Here is the documentation for it:
https://api-platform.com/docs/core/graphql/#queries

**Comment by BernardA** [comment]
@dunglas I have mentioned the repository above. [Here's the link again.](https://github.com/BernardA/quiamo-api).
@mahmoodbazdar  Thanks. That would explain why removing the `graphql` annotation from User and Address corrected the issue. It seems hard to follow this type of change if you are not specifically looking for it. 
But the Category entity did not have any specific `graphql` annotation to start with. I attempted to add the annotation as you indicated above to see what happens and nothing happened, ie, the problem persists.
Note that Category queries were working fine in `2.4.6` as the code stands. Now, as pointed out above, I can only get a simple list. No `edges` or `totalCount` available.

**Comment by BernardA** [comment]
Showing how Category looks on the `query` tab of `graphiql`. 
Normally it should show a `CategoryConnection` for categories as per the other resources.

      category(id: ID!): Category
       categories(
            title: String
            root: String
            root_list: [String]
            parent: String
            parent_list: [String]
            _order: CategoryFilter__order
        ): [Category]

**Comment by mahmoodbazdar** [comment]
@BernardA I checked your repo and the `edges`  and `totalCount ` is not available for `Category` entity because you disabled pagination.

https://github.com/BernardA/quiamo-api/blob/65c8e5369e52b759ce7556fb6602d79ddae3844e/src/Entity/Category.php#L48

**Comment by BernardA** [comment]
So, no pagination no edges or totalCount. Nice to know. I was not aware of that nor did I find something to that effect in the docs.
As it happens I made that change at the same time as the update to 2.5.0, so it became muddled up.
In the specific case of Category I need all of it to be available from initial page load and I will with the modified query, just with a different structure. That's fine.
